src/TestSuite/Constraint/Email/MailSubjectContains.php

```diff
@@ -0,0 +1,82 @@
+<?php
+declare(strict_types=1);
+
+/**
+ * CakePHP(tm) : Rapid Development Framework (https://cakephp.org)
+ * Copyright (c) Cake Software Foundation, Inc. (https://cakefoundation.org)
+ *
+ * Licensed under The MIT License
+ * For full copyright and license information, please see the LICENSE.txt
+ * Redistributions of files must retain the above copyright notice.
+ *
+ * @copyright     Copyright (c) Cake Software Foundation, Inc. (https://cakefoundation.org)
+ * @link          https://cakephp.org CakePHP(tm) Project
+ * @since         4.1.2
+ * @license       https://opensource.org/licenses/mit-license.php MIT License
+ */
+namespace Cake\TestSuite\Constraint\Email;
+
+/**
+ * MailSubjectContains
+ *
+ * @internal
+ */
+class MailSubjectContains extends MailConstraintBase
+{
+
+    /**
+     * Checks constraint
+     *
+     * @param mixed $other Constraint check
+     * @return bool
+     */
+    public function matches($other): bool
+    {
+        $messages = $this->getMessages();
+        foreach ($messages as $message) {
+            $subject = $message->getOriginalSubject();
+
+            $other = preg_quote($other, '/');
```

Comment:
You let "mixed" enter, but use it blindly as string. This can blow up
Maybe you want to either only accept string or assert it.

Reply:
I tried to change the method declaration to `public function matches(string $other): bool` it and got:
"bool should be compatible with PHPUnit\Framework\Constraint\Constraint::matches($other)"
what do you mean by "or assert it"

Reply:
Then dont change signature, but add assert inside:

If !is_string throw exception